packages/shared/src/graphql/sources.ts

```diff
@@ -1,18 +1,64 @@
 import { gql } from 'graphql-request';
 
+export enum SourceType {
+  Machine = 'machine',
+  Squad = 'squad',
+}
+
 export interface Source {
   __typename?: string;
   id?: string;
   name: string;
   image: string;
   handle: string;
+  type: SourceType;
+  permalink: string;
 }
 
-export const getSourcePermalink = (id: string): string =>
-  `${process.env.NEXT_PUBLIC_WEBAPP_URL}sources/${id}`;
+const SOURCE_PATH: Record<SourceType, string> = {
+  machine: 'source',
+  squad: 'squads',
+};
+
+export const getSourcePermalink = (id: string, type: SourceType): string => {
```

Comment:
I think if we just default it to `machine` type we support the notification concern right?

Reply:
Yes, but what if the notification is `squad` - we won't have any reference to say it is `squad`

Reply:
We don't need it if it's unknown right?
Only notifications have that, which is fine as they default to targetURL anyway